core/sql/optimizer/NATable.cpp

```diff
@@ -5485,9 +5485,13 @@ NABoolean createNAFileSets(TrafDesc * table_desc       /*IN*/,
 
    Lng32 postCreateNATableWarnings = CmpCommon::diags()->getNumber(DgSqlCode::WARNING_);
 
+
    if(postCreateNATableWarnings != preCreateNATableWarnings)
      tableConstructionHadWarnings_=TRUE;
-   const char *lobHdfsServer = CmpCommon::getDefaultString(LOB_HDFS_SERVER);
+   char  lobHdfsServer[256] ; // max length determined by dfs.namenode.fs-limits.max-component-length(255)
+   memset(lobHdfsServer,0,256);
+   strncpy(lobHdfsServer,CmpCommon::getDefaultString(LOB_HDFS_SERVER), sizeof(lobHdfsServer));
```

Comment:
If strlen(CmpCommon::getDefaultString(LOB_HDFS_SERVER)) is greater than or equal to sizeof(lobHdfsServer), then lobHdfsServer will not be null-terminated. So you might get some garbage characters on the end.

Consider nulling out the last byte. (You might want to make your buffer 256+1 bytes if the maximum length supported is 256.)

If strlen(CmpCommon::getDefaultString(LOB_HDFS_SERVER)) is greater than 256 bytes, this code will truncate it. Is that OK?

If not, consider using a dynamically sized array instead.